Sources/Purchasing/Purchases/TransactionPoster.swift

```diff
@@ -270,6 +249,30 @@ private extension TransactionPoster {
         }
     }
 
+    func fetchEncodedReceipt(transaction: StoreTransactionType,
+                             completion: @escaping (Result<EncodedAppleReceipt, BackendError>) -> Void) {
+        if systemInfo.dangerousSettings.internalSettings.usesStoreKit2JWS,
+           let jwsRepresentation = transaction.jwsRepresentation {
+            if transaction.environment == .xcode, #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) {
+                _ = Task<Void, Never> {
+                    completion(.success(.sk2receipt(await transactionFetcher.receipt)))
+                }
+            } else {
+                    completion(.success(.jws(jwsRepresentation)))
+            }
+        } else {
+            self.receiptFetcher.receiptData(
+                refreshPolicy: self.refreshRequestPolicy(forProductIdentifier: transaction.productIdentifier)
+            ) { receiptData, receiptURL in
+                if let receiptData = receiptData, !receiptData.isEmpty {
+                    completion(.success(.receipt(receiptData)))
+                } else {
+                    completion(.failure(BackendError.missingReceiptFile(receiptURL)))
+                }
+            }
+        }
+    }
+
```

Comment:
Extracted the logic to craft an `EncodedAppleReceipt` to its own method as it was getting complex